war/src/main/webapp/scripts/hudson-behavior.js

```diff
@@ -1767,11 +1767,11 @@ function scrollIntoView(e) {
     window.scrollBy(dx,dy);
 }
 
-// used in expandableTextbox.jelly to change a input field into a text area
+// used in expandableTextbox.jelly to change an input field into a text area
 function expandTextArea(button,id) {
     button.style.display="none";
     var field = button.parentNode.previousSibling.children[0];
-    var value = field.value.replace(/ +/g,'\n');
+    var value = field.value.replace('\n');
```

Comment:
I think this is intentional behaviour. Spaces are converted into new lines when expanding.

This sort of change will need an impact assessment to see how people are using it and whether this makes sense / won't break things.

(unless anyone believes differently)

Reply:
> I think this is intentional behaviour. Spaces are converted into new lines when expanding.

Oic, then it didn't work like that before, I guess. Though I'm not sure if that is more handy to work with, if you have an argument list with six or seven entries, you have to scroll down first to read thru it fully, instead of reading from the line start to end.

Reply:
I think it was for things like maven options, you could have them all one one line and they never had spaces in them. You could use this tool to make it easier to read them all at once even if it scrolled.

Reply:
Yep, used to work like Tim describes. Apparently it's been a while though -- 2.176.3 still does this, 2.190.3 does not.

Reply:
Seems to be down to something backported into 2.190.1. 2.190 is unaffected.

Reply:
Hmm interesting, so I guess we could just take this change?

seems like a reasonable change and if the previous behaviour was broken for years then 🤷

Reply:
https://github.com/jenkinsci/jenkins/commit/df600eb7a28dea189c41037398fd36fcea36302e perhaps?

Reply:
FWIW I'd prefer if the previous behavior was restored. Even if not used a lot, it's basically the reason this kind of form field exists.

Reply:
> [df600eb](https://github.com/jenkinsci/jenkins/commit/df600eb7a28dea189c41037398fd36fcea36302e) perhaps?

Yeah, restoring the behavior before the security was applied, resolves the issue.

Reply:
I submitted #6225 which would restore the previous behavior.